module/zfs/vdev_disk.c

```diff
@@ -585,7 +585,11 @@ __vdev_disk_physio(struct block_device *bdev, zio_t *zio,
 		/* Matching put called by vdev_disk_physio_completion */
 		vdev_disk_dio_get(dr);
 
+#ifndef HAVE_BIO_SET_DEV
 		dr->dr_bio[i]->bi_bdev = bdev;
+#else
+		bio_set_dev(dr->dr_bio[i], bdev);
+#endif
```

Comment:
This would be more readable if we defined our own version of `bio_set_dev()` in `linux/blkdev_compat.h` when `!HAVE_BIO_SET_DEV`.  A similar thing was done for `bio_set_op_attrs()`.

```c
#ifndef HAVE_BIO_SET_DEV
static inline void
bio_set_dev(struct bio *bio, struct block_device *bdev)
{
        bio->bi_bdev = bdev;
}
#endif /* !HAVE_BIO_SET_DEV */
```